File: typescript/food-ordering/app/src/restate-app/clients/payment_client.ts
/*
 * Copyright (c) 2023 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Restate Examples for the Node.js/TypeScript SDK,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/examples/blob/main/LICENSE
 */

class PaymentClient {
  async reserve(id: string, token: string, amount: number): Promise<boolean> {
    console.info(
      `[${id}] Reserving payment with token ${token} for $${amount}`
    );
    // do the call
    return true;
  }

  async unreserve(id: string, token: string, amount: number): Promise<boolean> {
    console.info(
      `[${id}] Unreserving payment with token ${token} for $${amount}`
    );
    // do the call
    return true;
  }

  async charge(id: string, token: string, amount: number): Promise<boolean> {
    console.info(
      `[${id}] Executing payment with token ${token} for $${amount}`
    );
    // do the call
    return true;
  }
}

export function getPaymentClient() {
  return new PaymentClient();
}

File: java/food-ordering/app/restate-app/src/main/java/dev/restate/sdk/examples/clients/PaymentClient.java
package dev.restate.sdk.examples.clients;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class PaymentClient {
    private static final Logger logger = LogManager.getLogger(PaymentClient.class);

    public static PaymentClient get(){
        return new PaymentClient();
    }

    public boolean reserve(String id, String token, double amount) {
        logger.info(String.format("[%s] Reserving payment with token %s for %.2f", id, token, amount));
        // do the call
        return true;
    }

    public boolean unreserve(String id, String token, double amount) {
        logger.info(String.format("[%s] Unreserving payment with token %s for %.2f", id, token, amount));
        // do the call
        return true;
    }

    public boolean charge(String id, String token, double amount) {
        logger.info(String.format("[%s] Executing payment with token %s for %.2f", id, token, amount));
        // do the call
        return true;
    }

}
